presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -346,7 +347,15 @@ public static ConnectorPageSource createParquetPageSource(
                         List<String> nestedColumnPath = nestedColumnPath(pushedDownSubfield);
                         Optional<ColumnIO> columnIO = findNestedColumnIO(lookupColumnByName(messageColumnIO, pushedDownSubfield.getRootName()), nestedColumnPath);
                         if (columnIO.isPresent()) {
-                            fieldsBuilder.add(constructField(type, columnIO.get()));
+                            if (isPushedDownSubfield(column)) {
+                                Type nestedType = getNestedStructType(pushedDownSubfield, type);
+                                GroupColumnIO newColumn = (GroupColumnIO) lookupColumnByName(messageColumnIO, pushedDownSubfield.getRootName());
```

Comment:
Can we add a check if it is type of GroupColumnIO? Just in case the leaf node get converted mistakenly.